From fba99f325920c02f96c9743c86928b388b97ab5c Mon Sep 17 00:00:00 2001
From: Jiri Olsa <jolsa@redhat.com>
Date: Mon, 8 Sep 2014 13:04:57 -0400
Subject: [tools] perf: Fix dwarf unwind max_stack processing

Message-id: <1410181752-25631-53-git-send-email-jolsa@redhat.com>
Patchwork-id: 91734
O-Subject: [PATCH RHEL7.1 BZ1134356 052/307] perf tools: Fix dwarf unwind max_stack processing
Bugzilla: 1134356
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>
RH-Acked-by: Don Zickus <dzickus@redhat.com>

From: Jiri Olsa <jolsa@kernel.org>

Bugzilla: 1134356
https://bugzilla.redhat.com/show_bug.cgi?id=1134356

upstream
========
commit b42dc32d4f91e4c0f34b628fdb012eb423da9e69
Author: Jiri Olsa <jolsa@redhat.com>
Date: Tue Jan 7 13:47:23 2014 +0100

description
===========
The 'unwind__get_entries' function currently returns 'max_stack + 1'
entries (instead of exact max_stack entries), because max_stack value
does not get decremented for the first entry.

This fix makes dwarf-unwind test pass.
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/tools/perf/util/unwind.c b/tools/perf/util/unwind.c
index 742f23b..bff3209 100644
--- a/tools/perf/util/unwind.c
+++ b/tools/perf/util/unwind.c
@@ -595,5 +595,5 @@ int unwind__get_entries(unwind_entry_cb_t cb, void *arg,
  if (ret)
   return -ENOMEM;
 
- return get_entries(&ui, cb, arg, max_stack);
+ return --max_stack > 0 ? get_entries(&ui, cb, arg, max_stack) : 0;
 }
-- 
1.7.1